app/services/platforms/tiktok.ts

```diff
@@ -0,0 +1,83 @@
+import { InheritMutations } from '../core';
+import { BasePlatformService } from './base-platform';
+import { IPlatformState, TPlatformCapability } from './index';
+import { IGoLiveSettings } from '../streaming';
+import { WidgetType } from '../widgets';
+import { assertIsDefined } from '../../util/properties-type-guards';
+
+export interface ITiktokStartStreamOptions {
+  serverUrl: string;
+  streamKey: string;
+}
+
+interface ITiktokServiceState extends IPlatformState {
+  settings: ITiktokStartStreamOptions;
+}
+
+@InheritMutations()
+export class TiktokService extends BasePlatformService<ITiktokServiceState> {
+  readonly apiBase = '';
+  readonly platform = 'tiktok';
+  readonly displayName = 'TikTok';
+  readonly capabilities = new Set<TPlatformCapability>([]);
+
+  // support only donation widgets for now
+  readonly widgetsWhitelist = [
+    WidgetType.AlertBox,
+    WidgetType.DonationGoal,
+    WidgetType.DonationTicker,
+    WidgetType.TipJar,
+  ];
```

Comment:
you can use the `platforms` field of `WidgetDisplayData` in `widgets-data` to accomplish this :)

Reply:
I found the `platforms` field for events there but don't see any example for widgets

Reply:
it currently only exists for platform-specific widgets, but can be extended to include all platforms except tiktok where needed
```
  [WidgetType.SupporterGoal]: {
    name: $t('Supporter Goal'),
    description: $t('Set a goal for your viewers to help you reach.'),
    demoVideo: false,
    demoFilename: 'source-follower-goal.png',
    supportList: [$t('Facebook Supporters')],
    platforms: new Set(['facebook']),
  },
```